### PR TITLE
Added Azure.KeyVault.RBAC

### LINKED ISSUE #1916: Consider using Azure role-based access control for key vaults
# Rule request

## Suggested rule change

Key vaults supports Azure role-based access control that is an authorization system built on Azure Resource Manager that provides fine-grained access management.

Benefits to customers:

- Access control by using the same API across Azure services.
- Centralized access management.
- Can be integrated with PIM for time-based access control. 
- The ability to have separate permissions on individual keys, secrets, and certificates.

The Azure RBAC also includes some disadvantages that should be documented to customers as this can be a key important decision maker for which authorization system they should use.

I think this would fit nice into the Security pillar.

## Applies to the following

The rule applies to the following:

- Resource type: **[Microsoft.KeyVault/vaults]**

## Additional context

[Provide access to Key Vault keys, certificates, and secrets with an Azure role-based access control](https://learn.microsoft.com/azure/key-vault/general/rbac-guide)
[Template reference](https://learn.microsoft.com/azure/templates/microsoft.keyvault/vaults)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #1916 

Added Azure.KeyVault.RBAC rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite Let me know.

File: docs/en/rules/Azure.KeyVault.RBAC.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+severity: Awareness
+pillar: Security
```
Comment (BernieWhite) [inline docs/en/rules/Azure.KeyVault.RBAC.md:3]:
```suggestion
pillar: Security
```

File: docs/en/rules/Azure.KeyVault.RBAC.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+### Configure with Azure PowerShell
+
+```powershell
+Update-AzKeyVault -ResourceGroupName '<resource_group>' -Name '<name>' -EnableRbacAuthorization
+```
+
+## NOTES
+
+The Azure RBAC permission model has disadvantages when compared to access policies, that might effect e.g infrastructure as code deployments. There are scenarios where Azure RBAC might not be suitable. More about this in the `LINKS` section.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.KeyVault.RBAC.md:93]:
```suggestion
The RBAC permission model may not be suitable for all use cases.
If this rule is not suitable for your use case, you can exclude or suppress the rule.
For information about limitations see _Azure role-based access control vs. access policies_ in the `LINKS` section.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Great work @BenjaminEngeset. Just a few minor changes suggested. See below.

**Review by BernieWhite — APPROVED** [review]
Great work @BenjaminEngeset. All good to merge.

**Pull request merged**